### ISSUE TITLE
Test: obj_memcheck/TEST0: SETUP (check/non-pmem/debug/memcheck) fails

### ISSUE THREAD

**Issue by Greg091** [pmem/issues#1020]
Environment Information:
- PMDK package version(s): 1.5.1-201-g6f13020b8
- OS(es) version(s): Fedora 29
- ndctl version(s):  64.1
- kernel version(s): 4.18.16-300.fc29.x86_64

Log of reproduction of the bug:
[obj_memcheck.txt](https://github.com/pmem/issues/files/2884003/obj_memcheck.txt)

**Comment by marcinslusarz** [comment]
Relevant part:
```
==5753== 12 bytes in 1 blocks are definitely lost in loss record 1 of 6
==5753==    at 0x483AB9D: realloc (vg_replace_malloc.c:826)
==5753==    by 0x4987AE4: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4989605: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498F630: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4982168: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498921E: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498F5A0: ndctl_pfn_get_first (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498F9B8: ndctl_namespace_get_pfn (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4858176: os_dimm_region_namespace (os_dimm_ndctl.c:157)
==5753==    by 0x48582DE: os_dimm_interleave_set (os_dimm_ndctl.c:194)
==5753==    by 0x48586CC: os_dimm_usc (os_dimm_ndctl.c:321)
==5753==    by 0x4869B73: shutdown_state_add_part (shutdown_state.c:94)
==5753== 
==5753== 12 bytes in 1 blocks are definitely lost in loss record 2 of 6
==5753==    at 0x483AB9D: realloc (vg_replace_malloc.c:826)
==5753==    by 0x4987AE4: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4989605: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498F630: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4982168: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498921E: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498F5A0: ndctl_pfn_get_first (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498F9B8: ndctl_namespace_get_pfn (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4858176: os_dimm_region_namespace (os_dimm_ndctl.c:157)
==5753==    by 0x48582DE: os_dimm_interleave_set (os_dimm_ndctl.c:194)
==5753==    by 0x485843D: os_dimm_uid (os_dimm_ndctl.c:230)
==5753==    by 0x4869BC1: shutdown_state_add_part (shutdown_state.c:99)
==5753== 
==5753== 12 bytes in 1 blocks are definitely lost in loss record 3 of 6
==5753==    at 0x483AB9D: realloc (vg_replace_malloc.c:826)
==5753==    by 0x4987AE4: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4989605: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498F630: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4982168: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498921E: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498F5A0: ndctl_pfn_get_first (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x498F9B8: ndctl_namespace_get_pfn (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4858176: os_dimm_region_namespace (os_dimm_ndctl.c:157)
==5753==    by 0x48582DE: os_dimm_interleave_set (os_dimm_ndctl.c:194)
==5753==    by 0x485843D: os_dimm_uid (os_dimm_ndctl.c:230)
==5753==    by 0x4869C66: shutdown_state_add_part (shutdown_state.c:112)
==5753== 
==5753== 108 bytes in 9 blocks are definitely lost in loss record 4 of 6
==5753==    at 0x483AB9D: realloc (vg_replace_malloc.c:826)
==5753==    by 0x4987AE4: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4989605: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4990870: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4982168: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x49892CE: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x49907E3: ndctl_dax_get_first (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4990C08: ndctl_namespace_get_dax (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x485805E: os_dimm_region_namespace (os_dimm_ndctl.c:131)
==5753==    by 0x48582DE: os_dimm_interleave_set (os_dimm_ndctl.c:194)
==5753==    by 0x48586CC: os_dimm_usc (os_dimm_ndctl.c:321)
==5753==    by 0x4869B73: shutdown_state_add_part (shutdown_state.c:94)
==5753== 
==5753== 108 bytes in 9 blocks are definitely lost in loss record 5 of 6
==5753==    at 0x483AB9D: realloc (vg_replace_malloc.c:826)
==5753==    by 0x4987AE4: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4989605: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4990870: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4982168: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x49892CE: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x49907E3: ndctl_dax_get_first (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4990C08: ndctl_namespace_get_dax (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x485805E: os_dimm_region_namespace (os_dimm_ndctl.c:131)
==5753==    by 0x48582DE: os_dimm_interleave_set (os_dimm_ndctl.c:194)
==5753==    by 0x485843D: os_dimm_uid (os_dimm_ndctl.c:230)
==5753==    by 0x4869BC1: shutdown_state_add_part (shutdown_state.c:99)
==5753== 
==5753== 108 bytes in 9 blocks are definitely lost in loss record 6 of 6
==5753==    at 0x483AB9D: realloc (vg_replace_malloc.c:826)
==5753==    by 0x4987AE4: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4989605: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4990870: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4982168: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x49892CE: ??? (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x49907E3: ndctl_dax_get_first (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x4990C08: ndctl_namespace_get_dax (in /usr/lib64/libndctl.so.6.13.0)
==5753==    by 0x485805E: os_dimm_region_namespace (os_dimm_ndctl.c:131)
==5753==    by 0x48582DE: os_dimm_interleave_set (os_dimm_ndctl.c:194)
==5753==    by 0x485843D: os_dimm_uid (os_dimm_ndctl.c:230)
==5753==    by 0x4869C66: shutdown_state_add_part (shutdown_state.c:112)
```

**Comment by Greg091** [comment]
Log of reproduction of the bug when ndctl is installed from source:
[memcheck0.log](https://github.com/pmem/issues/files/2885034/memcheck0.log)

**Comment by Greg091** [comment]
Verified on: 1.5.1-242-g1d9df9dfc + patch -> ndctl: fix memory leak in libndctl #84
